jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/MainActivity.java

```diff
@@ -96,11 +118,37 @@ public void sendMessage(String message) {
         // If the display message fragment is visible show the message there.
         if (displayMessageFragment != null) {
             displayMessageFragment.updateMessage(message);
-            return;
+        } else {
+            Log.w(MAIN_ACTIVITY_TAG,
+                    "Trying to send message, but displayMessageFragment is null.");
+        }
+
+        if (messageFile == null) {
```

Comment:
I agree that the check `messageFile == null` probably does not belong in this part of your code.

Yes, I think you will be good with simply removing this. :)